File: problems-extend-adapt/0001-0100/0026_drop-extra-copies/solution.py
from typing import List


class Solution:
    def dropExtraCopies(self, nums: List[int]) -> List[int]:
        # Sorted order puts every duplicate run adjacent, so one forward
        # scan can compact the array in place: write marks the end of the
        # unique prefix built so far, and the first element is always kept.
        write = 1
        for read in range(1, len(nums)):
            # nums[write - 1] is the last value kept; in a sorted array the
            # scan meets a new value exactly when the previous run ends.
            if nums[read] != nums[write - 1]:
                nums[write] = nums[read]
                write += 1
        # The statement frees the tail beyond the unique prefix, so the
        # compacted prefix is the whole judged answer; its length is k.
        return nums[:write]

File: problems-extend-adapt/0001-0100/0026_drop-extra-copies/solution.go
// Read/write two-pointer compaction: sortedness puts every duplicate run
// adjacent, so write can grow a unique prefix in place while read scans.
func dropExtraCopies(nums []int) []int {
	// The first element is always kept, so both pointers start at 1.
	write := 1
	for read := 1; read < len(nums); read++ {
		// nums[write-1] is the last value kept; in a sorted array the scan
		// meets a new value exactly when the previous run ends.
		if nums[read] != nums[write-1] {
			nums[write] = nums[read]
			write++
		}
	}
	// The statement frees the tail beyond the unique prefix, so the
	// compacted prefix is the whole judged answer; its length is k.
	return nums[:write]
}
